src/Body.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Http\Client;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Allows special input as body
+ *
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+interface Body
+{
+    /**
+     * Returns a set of headers which is needed to correctly send the body
+     *
+     * Note: these headers get overwritten by headers manually passed to the client
+     *
+     * Content-Length is calculated automatically if possible
+     *
+     * @return array
+     */
+    public function getContentHeaders();
+
+    /**
+     * Convert data to a format which can be used to create a proper PSR-7 Stream
+     *
+     * @return string|StreamInterface
```

Comment:
We have already discussed this in another issue. Body objects would need a message factory to return a stream object. And that's one simple IF to check the input parameter.

The alternative is to remove body types entirely and use "generators" which can populate the request based on a specific input. It would certainly make the contract cleaner. And we could move that code to the utility package.